### ISSUE TITLE
toc.follow not respected: right sidebar does not scroll or show "passed" items

### ISSUE THREAD

**Issue by melissawm** [jbms/sphinx-immaterial#380]
With a vanilla set up of this theme, the `toc.follow` feature seems to not be respected. The right sidebar does not scroll when the page has a large number of items, and there is no difference between items that have "passed", active items and items further down in the page.

I am probably doing something wrong, but can't detect what it is from the docs.

Here's how that looks:

![immaterial](https://github.com/user-attachments/assets/4e1913e7-c090-4d36-9781-ae3dc0ca33c4)

And here's the source code: https://github.com/melissawm/minimalsphinx

Any help would be appreciated. Thanks!

**Comment by 2bndy5** [comment]
It looks like you're using firefox in the picture. Is this behavior also observed in chrome?

**Comment by melissawm** [comment]
Yes, it's exactly the same on chrome. Also, I do see the effect in the deployed documentation for sphinx-immaterial, which is why I was expecting this to work at all.

**Comment by 2bndy5** [comment]
Can you point me to the page in our docs? I haven't seen this behavior. I tried quickly reproducing this locally but was unsuccessful.

**Comment by melissawm** [comment]
Oh I mean in your docs it works correctly in my browser 😄 See, for example, here: https://sphinx-immaterial.readthedocs.io/en/latest/admonitions.html

I am happy to help debug and submit a PR. I know this is coming in some way from mkdocs-material and if you can point me to the entry point for this functionality I might be able to take a look. Inspecting the code, I see that your pages are rendered with each item in the sidebar with class `md-nav__link md-nav__sticky` - even when using toc.follow (or toc.sticky) I never get my elements to show `md-nav__sticky` so I assume something is going wrong there.

**Comment by 2bndy5** [comment]
Does adding `toc.sticky` feature relieve the issue?

**Comment by melissawm** [comment]
Unfortunately no, same behavior.

**Comment by 2bndy5** [comment]
Hmm, IIRC the [current implementation for the `toc.follow` feature](https://github.com/jbms/sphinx-immaterial/blob/main/src/assets/javascripts/components/toc/index.ts) is inherited from upstream mkdocs-material implementation. There might be some modifications on our end.

**Comment by melissawm** [comment]
Oh wow, I found it. It's the palette: if it's not set, the scrollbar does not appear and the names are not highlighted accordingly. Would it be ok if I checked to see if having a default palette be applied (when it is unset) would solve the issue?

**Comment by jbms** [comment]
Yes that sounds good, thanks.

**Comment by 2bndy5** [comment]
I looked for upstream issues that might be related. 
- squidfunk/mkdocs-material#4960

It might be worth adding a default value for the `palette` in https://github.com/jbms/sphinx-immaterial/blob/3c8fe16a499407a9a9b71b7dd2133c559cdccf95/sphinx_immaterial/__init__.py#L29-L41

**Comment by melissawm** [comment]
The other alternative would be to add a note to the documentation about this behavior.
